nova_vm/src/types/language/value.rs

```diff
@@ -0,0 +1,122 @@
+use crate::{
+    heap::{
+        ArrayHeapData, BigIntHeapData, Handle, NumberHeapData, ObjectHeapData, StringHeapData,
+        SymbolHeapData,
+    },
+    SmallInteger, SmallString,
+};
+
+/// 6.1 ECMAScript Language Types
+/// https://tc39.es/ecma262/#sec-ecmascript-language-types
+#[derive(Debug, Clone, Copy)]
+pub enum Value {
+    /// 6.1.1 The Undefined Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-undefined-type
+    Undefined,
+
+    /// 6.1.2 The Null Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-null-type
+    Null,
+
+    /// 6.1.3 The Boolean Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-boolean-type
+    Boolean(bool),
+
+    /// 6.1.4 The String Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-string-type
+    String(Handle<StringHeapData>),
+    SmallString(SmallString),
+
+    /// 6.1.5 The Symbol Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-symbol-type
+    Symbol(Handle<SymbolHeapData>),
+
+    /// 6.1.6.1 The Number Type
+    /// https://tc39.es/ecma262/#sec-ecmascript-language-types-number-type
+    Number(Handle<NumberHeapData>),
+    IntegerNumber(SmallInteger), // 56-bit signed integer.
+    FloatNumber(f32),
```

Comment:
**praise:** I thought about adding this as well but ended up not doing so due to thinking it wouldn't really work for the most part. Hiding in NaN and Infinities inside it is a good idea! Is a signed zero also representable in f32?

Reply:
Seems like it is so that can also be hidden in here! Nice!

Reply:
Oh yeah, I was also thinking that maybe we could implement our own `f56` :D But that may be quite hard to do and wouldn't necessarily bring much benefit.

Reply:
> Oh yeah, I was also thinking that maybe we could implement our own `f56` :D But that may be quite hard to do and wouldn't necessarily bring much benefit.

We can :)
Actually it may be rather trivial byte manipulation and modification. As for it not bringing much benefit, yes, it may be rather slow to resize f64 <-> f56 frequently (it's not as straightforward as bit-wise truncation).

Honestly once we get anywhere with this, I would love to experiment with this for the purpose of comparing against loading the heap floats instead, ... with some real scripts against this.